src/server/clients_info.rs

```diff
@@ -229,10 +271,20 @@ impl ClientInfo {
     /// Removes a despawned entity tracked by this client.
     pub fn remove_despawned(&mut self, entity: Entity) {
         self.ticks.remove(&entity);
+        self.visibility.remove_despawned(entity);
         // We don't clean up `self.updates` for efficiency reasons.
         // `Self::acknowledge()` will properly ignore despawned entities.
     }
 
+    /// Iterates over all entities for which visibility was lost during this tick and removes them.
+    ///
+    /// Removal happens lazily during the iteration.
+    pub(super) fn remove_lost_visibility(&mut self) -> impl Iterator<Item = Entity> + '_ {
+        self.visibility.iter_lost_visibility().inspect(|entity| {
+            self.ticks.remove(entity);
+        })
+    }
```

Comment:
If I drain it here, I won't be able to unset `bool` inside `list`s by directly accessing added or removed entities.
I will need to iterate over the whole list.
Maybe it's fine since these methods are called differently?

Reply:
Oh, right! It will work then, will do as you suggested. And add a comment of course about what is where cleared.